lua/telescope/pickers.lua

```diff
@@ -1544,11 +1544,24 @@ function pickers.on_close_prompt(prompt_bufnr)
           picker.manager = EntryManager:new(picker.max_results, picker.entry_adder, picker.stats)
         end
       end
-      picker.default_text = picker:_get_prompt()
+      local curr_prompt = picker:_get_prompt()
+      picker.default_text = curr_prompt
       picker.cache_picker.selection_row = picker._selection_row
-      picker.cache_picker.cached_prompt = picker:_get_prompt()
-      picker.cache_picker.is_cached = true
-      table.insert(cached_pickers, 1, picker)
+
+      if picker.cache_picker.ignore_empty_prompt then
+        -- Ignore empty prompts if `cache_picker.ignore_empty_prompt` is set to true
+        if not (curr_prompt == nil or curr_prompt == "") then
+          picker.cache_picker.cached_prompt = curr_prompt
+          table.insert(cached_pickers, 1, picker)
+
+          picker.cache_picker.is_cached = true
+        end
+      else
+        picker.cache_picker.cached_prompt = curr_prompt
+        table.insert(cached_pickers, 1, picker)
+
+        picker.cache_picker.is_cached = true
+      end
```

Comment:
As much as I hate the code duplication here, and I did consider the following:

```lua
if not picker.cache_picker.ignore_empty_prompt or (curr_prompt and curr_prompt ~= "") then
    picker.cache_picker.cached_prompt = curr_prompt
    table.insert(cached_pickers, 1, picker)
    picker.cache_picker.is_cached = true
end
```

But the above^ snippet was very hard to read/understand as too many conditions in an if condition can make it harder to debug later on in case of any bug. I'm open to suggestions here but thought I'll share what I have considered with this piece of code. :)

Reply:
i dont think `not picker.cache_picker.ignore_empty_prompt or (curr_prompt and curr_prompt ~= "")` is unreadable. i actually would argue its more readable than having the duplicated code